### DISCUSSION TITLE
Is there a way to make the return-path domain and From domain 100% match for any domains setup in Postal?

### DISCUSSION THREAD

**Discussion by JohnHammell** [postalserver/postal#3195] [Help with using Postal]
Is there a way to make the return-path domain and From domain always exact match for any domains setup in Postal?

The purpose with this would be able to do Strict DMARC Alignment via SPF and DKIM alignment versus the "or" alignment option. Apparently strict DMARC alignment is SPF and/or DKIM alignment. And we get DKIM strict alignment with Postal as long as the DKIM DNS record is setup correctly.

But would also like to get the return path matching the From domain as well so that SPF also strictly aligns.

Is this possible with some special setting for postal.yml's return_path_domain: config? Maybe something like "return_path_domain: use_from_domain"?

With the return_path_domain setting, I believe we can achieve strict DKIM & SPF for a single domain in Postal, then any others would have strict via the DKIM &/or SPF. For example, using example.com as the return_path_domain would strict align both DKIM & SPF for example.com, but for support.example.com SPF would be relaxed alignment since it would use the example.com return_path_domain as well. DKIM would still cover support.example.com strict alignment, but I am looking to do both if possible rather than either/or.

---

Edited to add in this additional question: Are there any issues with using a root domain, example.com, as the return_path_domain instead of something like: rp.mail.example.com ?

**Comment by willpower232** [comment]
I don't really know enough about DMARC and DKIM to say if its worth it but I guess the major problem with making them match is that Postal would end up receiving all your incoming email and then it might not be able to tell the difference between bounced messages and regular incoming messages.

**Reply by JohnHammell** [reply]
I'm not sure it's worth it just in terms of DMARC passing since DMARC strict alignment requires only one of DKIM or SPF to exact match domains with the From domain.

However, I've read that both DKIM and SPF exact matching domains with the From domain is or may be viewed by some email providers as being a more secure / higher level of trust. So, passing DMARC with one in strict alignment is good but passing DMARC with both DKIM & SPF in strict alignment is even better, for deliverability. I suppose it would be like getting bonus points due to both DKIM & SPF meeting the strict alignment requirement.

One of the nice things about Postal is that it can handle either just outbound or both outbound + inbound. Probably even just inbound as well if anyone wanted to use it that way for inbound mail filtering.

The way I will be using Postal with my new install is for both outbound + inbound. Inbound I will have filtered through the rspamd integration. And then Postal will relay any messages that filter through to my inbox mail server.

My guess is that Postal should be able to use the root domain, example.com, as the return-path & for bounced messages since SPF alignment checks the From & return-path domains and has both relaxed & strict alignment options. I'll post a quick question about this with the dev tagged to see if we can get clarification in how Postal handles this.

**Comment by JohnHammell** [comment]
@adamcooke  , quick question just to get clarification on Postal's return_path_domain functionality.

**Can Postal handle using example.com as the return_path_domain for a Postal Server which sends email From example.com as well as possibly From different subdomains,  subdomain.example.com ?**

Purpose of this is for SPF strict domain alignment at least with the root domain.
